src/arrayMethodSplice.js

```diff
@@ -5,7 +5,56 @@
  */
 function applyCustomSplice() {
   [].__proto__.splice2 = function(start, deleteCount, ...items) {
-    // write code here
+    const leftArr = [];
+    const rightArr = [];
+    const removedElements = [];
+
+    if (arguments.length < 1 || start > this.length || deleteCount < 0) {
+      return [];
+    }
+
+    let removedPoint = 0;
+    if (start < 0 && start > -this.length) {
+      removedPoint = this.length + start;
+    } else if (start < -this.length || start === undefined) {
+      removedPoint = 0;
+    } else {
+      removedPoint = start;
+    }
+
+    const removeStart = removedPoint;
```

Comment:
do you really need this variable?

Reply:
yes, because in the next loops I'm changing this 2 variables

Reply:
change some blocks order and you won't need it anymore